packages/components/src/card/types.ts

```diff
@@ -0,0 +1,87 @@
+/**
+ * External dependencies
+ */
+// eslint-disable-next-line no-restricted-imports
+import type { CSSProperties } from 'react';
+
+/**
+ * Internal dependencies
+ */
+import type { Props as SurfaceProps } from '../surface/types';
+
+export type SizeOptions = 'xSmall' | 'small' | 'medium' | 'large';
+
+export type Props = SurfaceProps & {
+	/**
+	 * Renders without a border.
+	 *
+	 * @default false
+	 */
+	isBorderless?: boolean;
+	/**
+	 * Renders with rounded corners.
+	 *
+	 * @default true
+	 */
+	isRounded?: boolean;
+	/**
+	 * Determines the amount of padding within the component.
+	 *
+	 * @default 'medium'
+	 */
+	size?: SizeOptions;
+	/**
+	 * Renders with elevation styles (box shadow).
+	 *
+	 * @default false
+	 */
+	isElevated?: boolean;
+};
+
+type BaseSubComponentProps = {
+	/**
+	 * The children elements.
+	 */
+	children: React.ReactNode;
+	/**
+	 * Renders with a light gray background color.
+	 *
+	 * @default false
+	 */
+	isShady?: boolean;
+	/**
+	 * Determines the amount of padding within the component.
+	 *
+	 * @default 'medium'
+	 */
+	size?: SizeOptions;
```

Comment:
Nit: should we have a `Sizable` type with a single `size` prop that both `BaseSubComponentProps` and `Props` could extend?

Reply:
Done in https://github.com/WordPress/gutenberg/pull/32561/commits/c67b11ac6453524a14b776eb3891e6ffcd4f19de